src/score.js

```diff
@@ -0,0 +1,61 @@
+const fs = require("fs");
+const scoreFile = "./score.json";
+
+async function processScore(botContext) {
+  const score = botContext.sessions[botContext.chatID].scoreInSession;
+  const scoreBoard = await readScoreFromFile();
+
+  if (botContext.chatID in scoreBoard) {
+    if (score > scoreBoard[botContext.chatID].hiScore) {
+      scoreBoard[botContext.chatID].hiScore = score;
+      botContext.hiScore = score;
+    }
+  } else {
+    scoreBoard[botContext.chatID] = { hiScore: score, userName: botContext.userName };
+    botContext.hiScore = score;
+  }
+  await saveScoreToFile(scoreBoard);
+}
+
+async function getHiScore(botContext) {
+  const scoreBoard = await readScoreFromFile();
+  if (botContext.chatID in scoreBoard) {
+    botContext.hiScore = scoreBoard[botContext.chatID].hiScore;
+  } else {
+    scoreBoard[botContext.chatID] = { hiScore: 0, userName: botContext.userName };
+    botContext.hiScore = 0;
```

Comment:
It will be better to avoid mutations at all

Reply:
I don’t understand what you mean. What other mutations? What are mutations?